src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,64 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size;
+
+    public MateStack() {
+        stack = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        stack[size++] = value;
+        resize();
     }
 
     public T peek() {
-        return null;
+        T value = getValue(size);
+        checkToEmpty(value);
+        return value;
     }
 
     public T pop() {
-        return null;
+        T oldValue = getValue(size);
+
+        checkToEmpty(oldValue);
+
+        T[] oldStack = stack;
+        stack = (T[]) new Object[size - 1];
+        System.arraycopy(oldStack, 0, stack, 0, --size);
+        return oldValue;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size == stack.length) {
+            T[] oldStack = stack;
+            stack = (T[]) new Object[size + 1];
```

Comment:
hmm... I would say your stack will be with a bit lower performance if you resize it with +1, you can add at least 50% of stack.length

Reply:
done